src/OpenRoad.cc

```diff
@@ -378,17 +384,21 @@ void OpenRoad::writeAbstractLef(const char* filename,
   if (!block) {
     logger_->error(ORD, 53, "No block is loaded.");
   }
+  std::string tmp_filename = utl::createTmpFileName(filename);
+
   std::ofstream os;
   os.exceptions(std::ofstream::badbit | std::ofstream::failbit);
-  os.open(filename);
+  os.open(tmp_filename);
   odb::lefout writer(logger_, os);
   writer.setBloatFactor(bloat_factor);
   writer.setBloatOccupiedLayers(bloat_occupied_layers);
   writer.writeAbstractLef(block);
+  fs::rename(tmp_filename, filename);
```

Comment:
shouldn't this check for failure and report an error accordingly?

Reply:
Doesnt it throw an exception? It is 2024 😵‍💫

Reply:
It does https://en.cppreference.com/w/cpp/filesystem/rename

Reply:
Thank you, I'll add the exception handling.

Reply:
No need to add exception handling, it is already there.

To be a bit more specific, no exception handling:

1. check if the destination file exists, if it exists, then delete it. If deleting fails, then this should throw an exception.
2. open the .tmp file in overwrite mode alongside the final .odb file. If this fails, throw exception. Nothing to be done and the exception is handled higher up.
3. rename .tmp to final destination file. If this fails, then an exception is thrown and it is handled higher up. Nothing for the code to do here.